features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java

```diff
@@ -235,9 +236,42 @@ public Response getDeviceConfigsByInterface(Integer ipInterfaceId, String config
 
     /** {@inheritDoc} */
     @Override
-    public void deleteDeviceConfig(long id) {
-        throw new UnsupportedOperationException("Delete device config not yet supported.");
-        // deviceConfigDao.delete(id);
+    public Response deleteDeviceConfigs(List<Long> ids) {
+        if (CollectionUtils.isEmpty(ids)) {
+            LOG.debug("Bad request : empty or null DeviceConfig Id");
+            return Response.status(Status.BAD_REQUEST).entity("Invalid 'id' parameter").build();
+        }
+        sessionUtils.withTransaction(() -> {
+            try {
+                final List<DeviceConfig> deviceConfigList = ids.stream()
+                        .map(deviceConfigDao::get)
```

Comment:
Can we return 404 if any ID was not found? Otherwise this can return `null` and result in a NPE in the delete call

Reply:
code changed to return 404 (transaction code also changed since earlier implemention was returning null only)